doc/specs/#5000 - Process Model 2.0/#1032 - Elevation Quality of Life Improvements.md

```diff
@@ -0,0 +1,385 @@
+requested profile, do we always want to just create a new split? If the command
+ends up glomming to an existing window, does that even make sense? That invoking
+an elevated split in an unelevated window would end up splitting the elevated
+window? It's very possible that the user wanted a split in the tab they're
+currently in, in the unelevated window, but they don't want a split in the
+elevated window.
+
+What if there's not space in the elevated window to create the split (but there
+would be in the current window)? That would sure make it seem like nothing
+happened, silently.
+
+We could alternatively have cross-elevation splits default to always opening a
+new tab. That might mitigate some of the odd
+
+#### Configure the Terminal to _always_ run elevated
+
+`elevated` is a per-profile property, not a global property. If a user
+wants to always have all instances of the Terminal run elevated, they
+could set `"elevated": true` in their profile defaults. That would cause _all_
+profiles they launch to always spawn as elevated windows.
+
+#### `elevated` in Actions
+
+
+Additionally, we'll add the `elevated` property to the `NewTerminalArgs` used in
+the `newTab`, `splitPane`, and `newWindow` actions. This is similar to how other
+properties of profiles can be overriden at launch time. This will allow windows,
+tabs and panes to all be created specifically as elevated windows.
+
+We'll also add an iterable command for opening a profile in an
+elevated tab, with the following json:
+
+```jsonc
+{
+    // New elevated tab...
+    "name": { "key": "NewElevatedTabParentCommandName", "icon": "UAC-Shield.png" },
+    "commands": [
+        {
+            "iterateOn": "profiles",
+            "icon": "${profile.icon}",
+            "name": "${profile.name}",
+            "command": { "action": "newTab", "profile": "${profile.name}", "elevated": true }
+        }
+    ]
+},
+```
+
+## Implementation Details
+
+### Starting an elevated process from an unelevated process
+
+It seems that we're able to create an elevated process by passing the `"runas"`
+verb to
+[`ShellExecute`](https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutea).
+So we could use something like
+
+```c++
+ShellExecute(nullptr,
+             L"runas",
+             L"wt.exe",
+             L"-s -1 new-tab [args...]",
+             nullptr,
+             SW_SHOWNORMAL);
+```
+
+This will ask the shell to perform a UAC prompt before spawning `wt.exe` as an
+elevated process.
+
+### Starting an unelevated process from an elevated process
+
+As always, there's a blog post by Raymond Chen that describes in detail how we
+could go about launching a process _unelevated_ when we're currently elevated.
+For more details, please refer to [The Old New Thing: How can I launch an
+unelevated process from my elevated process, redux].
+
+## Potential Issues
+
+<table>
+<tr>
+<td><strong>Accessibility</strong></td>

```

Comment:
OMG did this table make it into the template... because if it didn't... it should.

Reply:
Not yet, but I'm hoping to update the template to re-organize some of the things, rename headers, etc. I'll merge this version of the table in for sure when I do that ☺️